exchanges/asset/asset.go

```diff
@@ -91,26 +110,72 @@ func (a Items) JoinToString(separator string) string {
 // IsValid returns whether or not the supplied asset type is valid or
 // not
 func (a Item) IsValid() bool {
-	for x := range supported {
-		if supported[x].String() == a.String() {
-			return true
-		}
+	return a != Empty && supportedFlag&a == a
+}
+
+// UnmarshalJSON comforms type to the umarshaler interface
+func (a *Item) UnmarshalJSON(d []byte) error {
+	var assetString string
+	err := json.Unmarshal(d, &assetString)
+	if err != nil {
+		return err
 	}
-	return false
+
+	if assetString == "" {
+		return nil
+	}
+
+	ai, err := New(assetString)
+	if err != nil {
+		return err
+	}
+
+	*a = ai
+	return nil
+}
+
+// MarshalJSON comforms type to the marshaller interface
+func (a Item) MarshalJSON() ([]byte, error) {
+	if a == Empty {
+		return json.Marshal("")
+	}
+	return json.Marshal(a.String())
```

Comment:
`a.String()` already has a case for a==Empty and returns `""`

Reply:
🦅